linux/LinuxProcessList.c

```diff
@@ -607,9 +612,14 @@ static bool LinuxProcessList_updateUser(ProcessList* processList, Process* proce
    return true;
 }
 
-static void LinuxProcessList_readIoFile(LinuxProcess* process, openat_arg_t procFd, unsigned long long realtimeMs) {
+static void LinuxProcessList_readIoFile(Process* proc, openat_arg_t procFd, bool scanMainThread, unsigned long long realtimeMs) {
```

Comment:
I think having both `LinuxProcessList_readStatFile` and `LinuxProcessList_readIoFile` take `LinuxProcess*` as the argument and accessing `process->super` to get the embedded `struct Process_` is a somewhat cleaner solution in my eyes. Otherwise your code assumes that you pass the right structure even if that's not a given; using a `LinuxProcess*` this is the responsibility of the caller. Also avoids the local variable (if you like).

Reply:
Changed